### PR TITLE
Fix for: #38  throws MessagingEntityNotFoundException if it can't find registration

### PR DESCRIPTION
**Pull request by JakubHolovsky**
#38  throws MessagingEntityNotFoundException if it can't find registration

I fixed the issue when MessagingEntityNotFoundException was being thrown if a registration was not found in "RegistrationExistsAsync" method. The root issue was caused by response.Content being NULL and therefore any operations on that property were failing.

### CODE REVIEW

**Comment by brannon** [comment]
@JakubHolovsky thanks for the PR! I'll take a look as soon as I can.

File: src/Microsoft.Azure.NotificationHubs/NotificationHubClient.cs [hunk 1]
```diff
@@ -2010,11 +2010,13 @@ private async Task DeleteRegistrationImplAsync(string registrationId, string eta
             using (var request = CreateHttpRequest(HttpMethod.Get, requestUri.Uri, out var trackingId))
             using (var response = await SendRequestAsync(request, trackingId, HttpStatusCode.OK, cancellationToken).ConfigureAwait(false))
             {
+                if (response.Content == null)
+                    return default;
```
Comment (brannon) [inline src/Microsoft.Azure.NotificationHubs/NotificationHubClient.cs]: @JakubHolovsky the CI build for this PR is failing on our Windows build agent with the following message:

```
NotificationHubClient.cs(2014,28): error CS8107: Feature 'default literal' is not available in C# 7.0. Please use language version 7.1 or greater. [D:\a\1\s\src\Microsoft.Azure.NotificationHubs\Microsoft.Azure.NotificationHubs.csproj]
```

While the `default` keyword is handy when the generic type isn't constrained, in this case `TEntity` is constrained as `where TEntity : EntityDescription`. I think it's cleaner and more clear to return `null` and that would avoid the build error as well.

Can you make this change to your PR? I'll check to make sure the change fixes the CI build.

Thanks!

Comment (JakubHolovsky) [reply src/Microsoft.Azure.NotificationHubs/NotificationHubClient.cs]: Hello, sure thing, changed default to null. Should be all good now.

**Comment by JakubHolovsky** [comment]
@brannon I updated the code to return null, Should be all good now.

**Comment by brannon** [comment]
@JakubHolovsky The unit tests pass, but I tried reproducing the problem with a test app. The test app still throws `MessagingEntityNotFoundException` when the registration does't exist. It's throwing in the call to `SendRequestAsync`, because the response code is a 404 (and only `200` is considered successful in this case).

`SendRequestAsync` has an overload where you can pass an array of expected status codes. You could use that overload to allow both `200` and `404`. My one hesitation there is that we use `GetEntityImplAsync` in several places and we need to make sure it still throws `MessagingEntityNotFoundException` for all of the other cases. You could add an optional parameter to `GetEntityImplAsync` for `bool throwIfNotFound = true` so that we don't change the existing default behavior.

btw- I made the unit tests fail by changing the response status code from `200` to `404` in the `null` Content case.

**Comment by JakubHolovsky** [comment]
Cheers @brannon , your suggestion makes sense, I modified the code according to your suggestion.

**Comment by brannon** [comment]
@JakubHolovsky thanks for the PR! I pushed some minor changes to you branch to fix an issue I found while testing the changes. Assuming the CI build passes, I'll merge this in.

**Pull request merged**